src/test/codegen/function-arguments.rs

```diff
@@ -139,6 +139,12 @@ pub fn return_slice(x: &[u16]) -> &[u16] {
   x
 }
 
+// CHECK: { i64, i64 } @enum_id(i64 %x.0, i64 %x.1)
+#[no_mangle]
+pub fn enum_id(x: Result<i64, i64>) -> Result<i64, i64> {
```

Comment:
Why does `Option` around this create problems? Does it require the combination of your 2 PRs? Can we wait to land one and actually have an `Option<Result<i64, i64>>` test too?

Reply:
My other PR landed, let me rebase this one.

Reply:
Indeed, with the combination of the two optimisations, `Option<Result<i64, i64>>` also ends up returned as `{ i64, i64 }`.